### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by utatti**
Some project-level refactorings.

- Use package.json `"files"` field instead of `.npmignore`
    - https://docs.npmjs.com/files/package.json#files
- Use [Prettier](https://github.com/prettier/prettier) instead of TSLint
    - As TypeScript compiler get rids of needs for linting, what's needed is a formatter, not a linter
    - Include a commit formatting existing source codes
- Update `.travis.yml`, update target versions of Node.js
    - Cut 4, add 8
- Add git hooks to format, build, and test before commit or push
    - Used [Husky](https://github.com/typicode/husky) to integrate with npm scripts

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,16 +7,28 @@
   },
   "main": "dist/index.js",
   "types": "types/index.d.ts",
+  "files": [
```
Comment (kawasako) [inline package.json]: I think that "LICENSE" and "README.md" are necessary to "files".

Comment (utatti) [reply package.json]: According to the package.json documentation, they will be included by default.

> Certain files are always included, regardless of settings:
>
> * package.json
> * README
> * CHANGES / CHANGELOG / HISTORY
> * LICENSE / LICENCE
> * NOTICE
> * The file in the "main" field

**Comment by kawasako** [comment]
Reviewed. I didn't know "Husky". It's very cool.

File: package.json (hunk 1 above)
Comment (kawasako) [reply package.json]: Oh, that's right. I get it.

**Comment by utatti** [comment]
Thanks for review. I'll merge this.

**Pull request merged**